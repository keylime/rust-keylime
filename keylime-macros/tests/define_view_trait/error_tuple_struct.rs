use keylime_macros::define_view_trait;

struct Target { inner: u8 }

#[define_view_trait(for_struct = "Target")]
struct BadView(u8);          // this is a tuple struct, should error

fn main() {}
