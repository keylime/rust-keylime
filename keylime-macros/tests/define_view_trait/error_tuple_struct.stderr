error: This macro only supports structs with named fields.
 --> tests/define_view_trait/error_tuple_struct.rs:5:1
  |
5 | #[define_view_trait(for_struct = "Target")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `define_view_trait` (in Nightly builds, run with -Z macro-backtrace for more info)
